File: shopping-cart-service/src/cart/presenters /controllers/cart.controller.ts
import { Controller, Post, Body, Param } from '@nestjs/common';
import { CommandBus } from '@nestjs/cqrs';
import { AddItemToCartCommand } from '../../application/commands/add-item-to-cart.command';

@Controller('api/carts')
export class CartController {
  constructor(private readonly commandBus: CommandBus) {}

  @Post(':userId/items')
  async addItemToCart(
    @Param('userId') userId: string,
    @Body() body: { productId: string; quantity: number },
  ): Promise<void> {
    const command = new AddItemToCartCommand(
      userId,
      body.productId,
      body.quantity,
    );
    await this.commandBus.execute(command);
  }
}

File: shopping-cart-service/src/main/java/com/carlosoliveira/ecommerce/shoppingcart/presentation/CartController.java
package com.carlosoliveira.ecommerce.shoppingcart.presentation;

import com.carlosoliveira.ecommerce.shoppingcart.application.commands.handlers.AddItemToCartCommandHandler;
import com.carlosoliveira.ecommerce.shoppingcart.application.commands.AddItemToCartCommand;
import com.carlosoliveira.ecommerce.shoppingcart.application.dtos.AddItemRequest;
import lombok.AllArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.UUID;

@RestController
@RequestMapping("/api/carts")
@AllArgsConstructor
public class CartController {

    private final AddItemToCartCommandHandler addItemToCart;

    @PostMapping("/{userId}/items")
    public ResponseEntity<Void> addItem(@PathVariable UUID userId, @RequestBody AddItemRequest request) {
        AddItemToCartCommand command = new AddItemToCartCommand(
                userId,
                request.productId(),
                request.productName(),
                request.price(),
                request.quantity()
        );
        addItemToCart.handle(command);
        return ResponseEntity.ok().build();
    }
}
